src/datablocks/datablock.go

```diff
@@ -21,22 +21,31 @@ type DataBlock struct {
 }
 
 func NewDataBlock(cols []columns.Column) *DataBlock {
-	var values []*DataBlockValue
-	indexmap := make(map[string]int)
-	valuesmap := make(map[string]*DataBlockValue)
-
-	for i, col := range cols {
-		cv := NewDataBlockValue(col)
-		indexmap[col.Name] = i
-		valuesmap[col.Name] = cv
-		values = append(values, cv)
-	}
-	return &DataBlock{
+	block := &DataBlock{
 		info:      &DataBlockInfo{},
-		values:    values,
-		indexmap:  indexmap,
-		valuesmap: valuesmap,
+		values:    []*DataBlockValue{},
+		indexmap:  make(map[string]int),
+		valuesmap: make(map[string]*DataBlockValue),
 	}
+
+	for _, col := range cols {
+		block.Insert(col)
+	}
+	return block
+}
+
+// Clone a sample block
+func (block *DataBlock) Clone() *DataBlock {
+	return NewDataBlock(block.Columns())
+}
+
+func (block *DataBlock) Insert(col columns.Column) {
```

Comment:
'InsertColumn' is better here?

Reply:
Right, Golang doesn't have function overload.